src/lib/models/status_history.rs

```diff
@@ -0,0 +1,81 @@
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+use serde_json::{from_value, json, Value as JsonValue};
+use strum_macros::{Display as EnumToString, EnumString};
+use time::OffsetDateTime;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, EnumToString, EnumString)]
+pub enum Status {
+    NotReported,
+    Reported,
+    WillNotReport,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct StatusHistoryEntry {
+    pub t: OffsetDateTime,
+    pub status: Status,
```

Comment:
I recall you mentioning at some point that you wanted the ability to add arbitrary metadata here. I was expecting to see some kind of "meta" HashMap here. Is that not relevant anymore/not possible?

Reply:
It's definitely possible. If figured we'd add it when we need it.